src/js/components/Diagram/stories/Animated.js

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+
+import { Box, Diagram, Grommet, grommet, Stack, Text } from 'grommet';
+import { Diamond } from 'grommet-icons';
+import { deepMerge } from 'grommet/utils';
+
+import { data } from './data';
+
+const customTheme = deepMerge(grommet, {
+  diagram: {
+    extend:
+      '@keyframes example { to { stroke-dashoffset: 0; } } path { stroke-dasharray: 500; stroke-dashoffset: 500; animation: example 3s linear forwards; }',
+  },
+});
+
+const connection = (fromTarget, toTarget, { color, ...rest } = {}) => ({
+  fromTarget,
+  toTarget,
+  anchor: 'vertical',
+  color: 'accent-4',
+  thickness: 'xsmall',
+  round: true,
+  type: 'direct',
+  ...rest,
+});
+
+const DiamondContainer = ({ carat, color, cut, align, id, name, textSize }) => (
+  <Box
+    align={align || 'center'}
+    gap="medium"
+    direction="row"
+    key={name}
+    alignSelf="center"
+  >
+    <Diamond id={id} size="xlarge" color="neutral-3" />
+    <Box align={align}>
+      <Text size="medium" weight="bold">
+        {name}
+      </Text>
+      {carat && <Text size={textSize}> Carat: {carat} </Text>}
+      {color && <Text size={textSize}> Color: {color} </Text>}
+      {cut && <Text size={textSize}> Cut: {cut} </Text>}
+    </Box>
+  </Box>
+);
+
+const Container = (node, index) => (
+  <DiamondContainer
+    carat={node.carat}
+    color={node.color}
+    cut={node.cut}
+    id={index}
+    key={node.name}
+    name={node.name}
+    textSize="small"
+  />
+);
+
+class Animated extends React.Component {
+  state = {
+    draw: true,
+  };
+
+  componentDidMount() {
+    this.timer = setInterval(() => {
+      const { draw } = this.state;
+      this.setState({
+        draw: !draw,
+      });
+    }, 3000);
+  }
+
+  render() {
+    const { draw } = this.state;
+    const connections = [];
+
+    if (draw) {
+      connections.push(connection('4', '1', { anchor: 'vertical' }));
+      connections.push(connection('4', '2', { anchor: 'vertical' }));
+      connections.push(connection('4', '3', { anchor: 'vertical' }));
+    }
+
+    return (
+      <Grommet theme={customTheme}>
+        <Box align="center">
+          <Box pad="large">
+            <Stack>
+              <Box>
```

Comment:
not sure we need another box here.

Reply:
Removing it will be messy. I need the Stack container to stretch across the full width of this Box, and for the DiamondContainers & Diagram to be contained and organized within it, so it made sense for me to use to control the layout better.